### PR TITLE
Do not push to pypi-integration on merge to dev in Python

### PR DESCRIPTION
**Pull request by diasdm**
Hey,
Merges to dev are failing to build because they try to push to pypi-integration.
Not sure if there is a better way to fix this but had implemented this in the branch I was using for mobtest.

### CODE REVIEW

**Comment by diasdm** [comment]
Tested in https://github.com/MOV-AI/metadata-static-analysis/actions/runs/3067251843

**Review by AlexFernandes-MOVAI — COMMENTED** [review]
I guess that we do not push to dev since we have the packages available in github artifacts.
do you confirm ?

**Comment by diasdm** [comment]
I'm not sure I understand the question... This is my understanding:
- `pypi-edge` - for production, does not support overwrite, pushes occur when GH release is no longer a pre-release
- `pypi-integration` - for release candidates, does not support overwrite, pushes occur in `main` and `releases/*` branches
- `pypi-experimental` - for the latest builds, supports overwrite, pushes occurs every time the pipeline is executed (with exception of GH release)

The issue I faced is that we are pushing to `pypi-integration` on merge to `dev`, which, since `pypi-integration` does not allow overwrite, makes the pipeline fail.
So my solution was to prevent pushes to `pypi-integration` when the branch is `dev`.

File: .github/workflows/py-workflow.yml [hunk 1]
```diff
@@ -170,7 +170,7 @@ jobs:
       run: python3 -m pytest
 
     - name: Raise version
-      if: ${{ inputs.deploy == 'true' }}
+      if: ${{ (inputs.deploy == 'true') && (github.ref != 'refs/heads/dev') }}
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/py-workflow.yml:173]: `inputs.deploy` is already doing the triggering, so if we do not want to deploy on actions on `dev` branches, the configuration has to be done in each repo

**Comment by AlexFernandes-MOVAI** [comment]
Will be handled in another way :
- we do not want to build artifact twice
- deploy argument is already doing the customization of behavior
- if we want to modify the behavior, let's do it in the templates and the repos

**Pull request closed without merging**